hbase-server/src/test/java/org/apache/hadoop/hbase/mapreduce/TestTableInputFormatScanBase.java

```diff
@@ -271,6 +272,11 @@ public void testNumOfSplits(int splitsPerRegion, int expectedNumOfSplits) throws
     TableInputFormat tif = new TableInputFormat();
     tif.setConf(job.getConfiguration());
     List<InputSplit> splits = tif.getSplits(job);
+    for (InputSplit split : splits) {
+      TableSplit tableSplit = (TableSplit) split;
+      Assert.assertEquals(tableSplit.getScan().getStartRow(), HConstants.EMPTY_START_ROW);
```

Comment:
nit: I think these checks are not tight enough since they don't differentiate between a full table scan object and a no scan object, instead I think it should be something like

assertTrue(tableSplit.getScanAsString().isEmpty()) or some such...

Reply:
@bharathv The reason I am asserting on this is when we serialize null scan object, we set the default object. That's why we don't get the empty string back but with the defaut values. I am asserting here on default values.
https://github.com/apache/hbase/blob/branch-1/hbase-client/src/main/java/org/apache/hadoop/hbase/protobuf/ProtobufUtil.java#L1126

Reply:
Ya my point is, that is a weak check. You wouldn't know if that scan is an empty scan or a default scan. Change the code to something like this and you can see there is a bug ...

 ```
for (InputSplit split : splits) {
      TableSplit tableSplit = (TableSplit) split;
      assertTrue(tableSplit.getScanAsString().isEmpty());
  }

In table split code, add this...

 public String getScanAsString() {
    return scan;
  }
```

Run this test

```
  @Test
  public void testGetSplits() throws IOException, InterruptedException, ClassNotFoundException {
    testNumOfSplits(1, 26);
    testNumOfSplits(3, 78);  <====
  }
```

If number of splits per region is > 1, your patch didn't handle the TableSplit constructor in this call createNInputSplitsUniform(). Your test still passes because you are asserting on a default Scan object and not the fact that it shouldn't be set in the first place.

Reply:
I see your point, thanks !